test/functional/tests/fault_injection/test_fault_injection_many_to_one.py

```diff
@@ -0,0 +1,254 @@
+
+    with TestRun.step("Add both core devices to cache."):
+        core1 = cache.add_core(core_part1)
+        core2 = cache.add_core(core_part2)
+        cores_count = len(casadm_parser.get_cores(cache.cache_id))
+        if cores_count != 2:
+            TestRun.fail(f"Expected cores count: 2; Actual cores count: {cores_count}.")
+
+    with TestRun.step("Change sequential cutoff policy."):
+        cache.set_seq_cutoff_policy(SeqCutOffPolicy.never)
+
+    with TestRun.step("Fill cache with pages from the first core."):
+        blocks = int(cache.size.value / block_size.value)
+        if cache_mode != CacheMode.WO:
+            dd = (Dd()
+                  .input(core1.system_path)
+                  .output("/dev/null")
+                  .block_size(block_size)
+                  .count(blocks))
+            dd.run()
+        elif cache_mode == CacheMode.WO:
+            dd = (Dd()
+                  .input("/dev/urandom")
+                  .output(core1.system_path)
+                  .block_size(block_size)
+                  .count(blocks))
+            dd.run()
+        core1_occupied_blocks_before = core1.get_occupancy()
+
+    with TestRun.step("Check if the remaining core is able to use cache."):
+        try:
+            if cache_mode != CacheMode.WO:
+                dd = (Dd()
+                      .input(core2.system_path)
+                      .output("/dev/null")
+                      .block_size(block_size)
+                      .count(blocks_100MiB))
+                dd.run()
+                os_utils.sync()
+                os_utils.drop_caches()
+                dd = (Dd()
+                      .input(core2.system_path)
+                      .output("/dev/null")
+                      .block_size(block_size)
+                      .count(blocks_100MiB))
+                dd.run()
+            elif cache_mode == CacheMode.WO:
+                dd = (Dd()
+                      .input("/dev/urandom")
+                      .output(core2.system_path)
+                      .block_size(block_size)
+                      .count(blocks_100MiB))
+                dd.run()
+                os_utils.sync()
+                os_utils.drop_caches()
+                dd = (Dd()
+                      .input("/dev/urandom")
+                      .output(core2.system_path)
+                      .block_size(block_size)
+                      .count(blocks_100MiB))
+                dd.run()
+            core1_occupied_blocks_after = core1.get_occupancy()
+        except Exception:
+            TestRun.LOGGER.error("The remaining core is not able to use cache.")
+
+    with TestRun.step("Check if occupancy from first core is removed from cache."):
+        # The first core's occupancy should be lower than cache's occupancy
+        # by the value of the remaining core's occupancy because cache
+        # should reallocate blocks from unused core to used core.
+        if core1_occupied_blocks_after >= core1_occupied_blocks_before:
```

Comment:
I believe you wanted to check if cache1 occupancy is greater than core1 occupancy. I would also test if core2 occupancy is greater than 0.